File: api/inventory/definitions/inventory.ts
export interface ServerInventoryProvider {
    addItem(playerId: number, itemName: string, amount: number): boolean;
    removeItem(playerId: number, itemName: string, amount: number): boolean;
    getItemCount(playerId: number, itemName: string): number;
    hasItem(playerId: number, itemName: string, amount: number): boolean;
}

File: api/inventory/definitions/inventory.lua
---@alias InventoryType "player" | "stash" | "trunk" | "glovebox"

---@class InventoryRef
---@field type InventoryType
---@field id integer | string

---@class ServerInventoryProvider
---@field addItem fun(inventoryRef: InventoryRef, itemName: string, amount: number): boolean
---@field removeItem fun(inventoryRef: InventoryRef, itemName: string, amount: number): boolean
---@field getItemCount fun(inventoryRef: InventoryRef, itemName: string): number
---@field hasItem fun(inventoryRef: InventoryRef, itemName: string, amount: number): boolean
---@field canCarryItem fun(inventoryRef: InventoryRef, itemName: string, amount: number): boolean
